rw_mutex.go

```diff
@@ -101,24 +104,52 @@ func (m *RWMutex) RLock() error {
 	}
```

Comment:
Can we remove this loop above since it's in `tryToGetReadLock`?

Reply:
good catch
0aa9532b46656656ffa10b5e2ab47a1a266f8b8d